### PR TITLE
Webpack upgrade

### PR DESCRIPTION
**Pull request by veegandhi**
## Description

Upgrade webpack and dev, dist and test pipeline

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -22,14 +22,14 @@
...
-        "config": "config/dev.js"
+        "config": "config/webpack.config.dev.js"
       }
     }
   },
   "rules": {
     "id-length": [
       2,
-      { "min": 3, "max": 30, "properties": "never", "exceptions": ["_", "cb"] }
+      { "min": 2, "max": 30, "properties": "never", "exceptions": ["_", "cb"] }
```
Comment (veegandhi) [inline .eslintrc]: Making this consistent

File: docs/components/Header/index.jsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import './styles.scss';
 
 const HeaderGraphics = () => (
   <span>
-    <SvgSymbol href="./docs/assets/svg-symbols.svg#logo" />
+    <SvgSymbol href="./assets/svg-symbols.svg#logo" />
```
Comment (veegandhi) [inline docs/components/Header/index.jsx]: Removed `docs` from all paths as dev server's base path is `docs`

File: docs/components/Contributors/index.jsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import _ from 'lodash';
-import 'isomorphic-fetch';
+import axios from 'axios';
```
Comment (veegandhi) [inline docs/components/Contributors/index.jsx]: `axios` to be consistent with all repos

File: scripts/start.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+// We attempt to use the default port but if it is busy, we offer the user to
+// run on a different port. `detect()` Promise resolves to the next free port.
+choosePort(HOST, DEFAULT_PORT)
+  .then(port => {
+    if (port == null) {
+      // We have not found a port.
+      return;
+    }
+    const protocol = process.env.HTTPS === 'true' ? 'https' : 'http';
+    const appName = paths.appPackageJson.name;
```
Comment (knilink) [inline scripts/start.js]: `appName` was undefined.
should be either `const appName = require(paths.appPackageJson).name;` or just `const appName = 'direct-ui';`.

**Comment by knilink** [comment]
go following message when during `npm run dist`
`Browserslist: caniuse-lite is outdated. `
not sure whether it's related to this PR but upgrading `@babel/preset-env` seemed fixing it

**Comment by veegandhi** [comment]
@knilink, @bchew - I was looking at the dependency tree and looks like we cannot do much about the `caniuse-lite` upgrade warning at this stage.I think `optimize-css-assets-webpack-plugin` is having an outdated dependency, but we are on latest. 

There is an open issue on [CRA](https://github.com/facebook/create-react-app/issues/6708) as well. We are using ejected setup from that repo for our build and I just noticed the grid is also having the same noise when we generate the dist.

**Comment by knilink** [comment]
probably need to set externals for the dist
https://github.com/Adslot/adslot-ui/blob/c02a491282e394060125983d0396a6b17bc5e3b1/config/dist.js#L52-L89

File: docs/components/SearchBar/index.jsx [hunk 5]
```diff
@@ -20,7 +20,7 @@ class SearchBarComponent extends React.Component {
         additionalClassNames={['adslot-ui-searchbar']}
         searchString={this.state.searchBarString}
         searchPlaceholder="Search for component."
-        searchIconHref="./docs/assets/svg-symbols.svg#search"
+        searchIconHref="/assets/svg-symbols.svg#search"
```
Comment (lightbringer1991) [inline docs/components/SearchBar/index.jsx]: `./assets/svg-symbols.svg#search` for consistency?

File: docs/examples/SearchBarExample.jsx [hunk 6]
```diff
@@ -40,7 +40,7 @@ const exampleProps = {
   <SearchBar
     searchString={this.state.searchBarString}
     searchPlaceholder="Enter a word or phrase to find matching items."
-    searchIconHref="./docs/assets/svg-symbols.svg#search"
+    searchIconhref="/assets/svg-symbols.svg"
```
Comment (lightbringer1991) [inline docs/examples/SearchBarExample.jsx]: `./assets/svg-symbols.svg`?

**Comment by veegandhi** [comment]
squashed commits

**Comment by bchew** [comment]
Merging :rocket:

**Pull request merged**